sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusSessionReceiver.java

```diff
@@ -127,6 +130,7 @@
                     logger.info("entityPath[{}]. sessionId[{}]. Did not a receive message within timeout {}.",
                         receiveLink.getEntityPath(), sessionId.get(), retryOptions.getTryTimeout());
                     cancelReceiveProcessor.onComplete();
+                    idleTimeOutCleanupOperation.accept(this);
```

Comment:
This creates a circular dependency on the parent class which I try to shy away from because it's messy architecture. 
I am confident the session receiver has other ways that you can tell it's stopped.